src/App.tsx

```diff
@@ -33,19 +33,41 @@ function App() {
     const [configStr, setConfigStr] = useState<string>("");
     const [viewRecipe, setViewRecipe] = useState<boolean>(true);
     const [viewConfig, setViewConfig] = useState<boolean>(true);
+    const [viewResults, setViewResults] = useState<boolean>(false);
     const [viewLogs, setViewLogs] = useState<boolean>(true);
+    const [runTime, setRunTime] = useState<number>(0);
 
-    const submitRecipe = async () => {
+    let start = 0;
+
+    async function sleep(ms: number): Promise<void> {
+        return new Promise((resolve) => setTimeout(resolve, ms));
+    };
+
+    const submitRecipe = async (useECS: boolean = false) => {
+        setResultUrl("");
         const firebaseRecipe = "firebase:recipes/" + selectedRecipe
         const firebaseConfig = "firebase:configs/" + selectedConfig;
-        const url = getSubmitPackingUrl(firebaseRecipe, firebaseConfig);
+        const url = getSubmitPackingUrl(firebaseRecipe, firebaseConfig, useECS);
         const request: RequestInfo = new Request(url, {
-            method: "POST",
+            method: useECS ? "GET" : "POST",
         });
+        start = Date.now();
```

Comment:
const start = Date.now();

Reply:
so start get's initialized with `let start = 0;` on line 40, which I needed to include or else it wasn't refreshing properly. I think this isn't a great pattern, maybe we don't need to display the range time anyways? open to suggestion

Reply:
make sense!